cristin-import/src/main/java/no/unit/nva/cristin/lambda/PublicationUpdater.java

```diff
@@ -177,26 +196,59 @@ private static PublicationInstance<? extends Pages> getPublicationInstance(Publi
     private static PublicationContext updatePublicationContext(PublicationRepresentations publicationRepresentations) {
         var existingPublicationContext = getPublicationContext(publicationRepresentations.getExistingPublication());
         var incomingPublicationContext = getPublicationContext(publicationRepresentations.getIncomingPublication());
+        return updatePublicationContext(existingPublicationContext, incomingPublicationContext);
+    }
+
+    private static PublicationContext updatePublicationContext(PublicationContext existingPublicationContext,
+                                                            PublicationContext incomingPublicationContext) {
         if (existingPublicationContext instanceof Event existingEvent && incomingPublicationContext instanceof Event incomingEvent) {
-            var existingPlace = existingEvent.getPlace();
-            var incomingPlace = incomingEvent.getPlace();
-            if (shouldBeUpdated(existingPlace, incomingPlace)) {
-                return new Event.Builder()
-                           .withLabel(nonNull(existingEvent.getLabel()) ? existingEvent.getLabel() : incomingEvent.getLabel())
-                           .withAgent(nonNull(existingEvent.getAgent()) ? existingEvent.getAgent() : incomingEvent.getAgent())
-                           .withTime(nonNull(existingEvent.getTime()) ? existingEvent.getTime() : incomingEvent.getTime())
-                           .withProduct(existingEvent.getProduct().orElse(incomingEvent.getProduct().orElse(null)))
-                           .withSubEvent(existingEvent.getSubEvent().orElse(incomingEvent.getSubEvent().orElse(null)))
-                           .withPlace(incomingPlace)
-                           .build();
-            } else {
-                return existingEvent;
-            }
+            return updateEvent(existingEvent, incomingEvent);
+        }
+        if (existingPublicationContext instanceof MediaContribution existingMediaContribution
+            && incomingPublicationContext instanceof MediaContribution incomingMediaContribution) {
+            return new MediaContribution.Builder()
+                       .withFormat(existingMediaContribution.getFormat())
+                       .withMedium(existingMediaContribution.getMedium())
+                       .withDisseminationChannel(updateDisseminationChannel(existingMediaContribution, incomingMediaContribution))
+                       .build();
         } else {
             return existingPublicationContext;
         }
     }
 
+    private static String updateDisseminationChannel(MediaContribution existingMediaContribution,
```

Comment:
Kind of misleading name when you are not actually updating, but rather returning the "most up to date" one.

```suggestion
    private static String getMostUpToDateDisseminationChannel(MediaContribution existingMediaContribution,
```

Reply:
"Most up to date" might be misleading as well... `getCorrect` or `getCurrent` might be better? Anyways, up to you if you want to do something with it